### ISSUE TITLE
GVFS support

### ISSUE THREAD

**Issue by bviktor** [gitbucket/gitbucket#1444]
> GVFS stands for Git Virtual File System. GVFS virtualizes the file system beneath your git repo so that git and all tools see a fully hydrated repo, but GVFS only downloads objects as they are needed.

This thing is rather new so I'm pretty sure things will change in the coming months pretty quickly, but this ticket will be good as a heads-up at least.

References:

https://github.com/Microsoft/GVFS/blob/master/Protocol.md
https://github.com/Microsoft/GVFS/blob/master/Readme.md
http://www.phoronix.com/scan.php?page=news_item&px=Microsoft-GVFS-Git-Filesystem
https://blogs.msdn.microsoft.com/visualstudioalm/2017/02/03/announcing-gvfs-git-virtual-file-system/

**Comment by aadrian** [comment]
-1 : Gitbucket depends on JGit (and the JVM) .

**Comment by bviktor** [comment]
Meaning?

**Comment by aadrian** [comment]
> Meaning?

No native dependencies.

**Comment by bviktor** [comment]
Who said it needs to be implemented natively?

**Comment by aadrian** [comment]
> Who said it needs to be implemented natively?

Well, than maybe this ticket needs be posted in the JGit issue tracker :) .

**Comment by bviktor** [comment]
I don't quite understand. You're implying GitBucket mustn't extend JGit's functionality in any way.

**Comment by aadrian** [comment]
> I don't quite understand. You're implying GitBucket mustn't extend JGit's functionality in any way.

GitBucket is a UI around a generic git library implementation like JGit. I'm not sure that the scope of this project would be so big to include protocols too.

Recently support for Git LFS was added https://gitbucket.github.io/gitbucket-news/gitbucket/2017/01/03/git-lfs-support.html 
However GitBucket didn't implement the protocol itself: adopted the feature only when it was usable from JGit.

Since many more projects depend on JGit (even Eclipse :) ), IMHO it's a much better place for protocol extensions.

**Comment by bviktor** [comment]
I see. Then it might indeed be a better place to ask. Nevertheless, I think this ticket could be a useful tracker bug for adding support IF and when JGit adds support.

**Comment by aadrian** [comment]
> I think this ticket could be a useful tracker bug for adding support IF and when JGit adds support.

Of course. If the protocol would work in JGit, there would be still needed the implementation of the UI and the required binding/handling/functionality in GitBucket.

**Comment by bviktor** [comment]
[GVFS will be renamed](https://github.com/Microsoft/GVFS/issues/72)
